app/controllers/v0/sessions_controller.rb

```diff
@@ -27,41 +29,48 @@ def new
     end
 
     def saml_logout_callback
-      logout_response = OneLogin::RubySaml::Logoutresponse.new(params[:SAMLResponse], saml_settings,
-                                                               raw_get_params: params)
-      logout_request  = SingleLogoutRequest.find(logout_response&.in_response_to)
+      saml_response = SAML::Responses::Logout.new(params[:SAMLResponse], saml_settings, raw_get_params: params)
 
-      errors = build_logout_errors(logout_response, logout_request)
-      additional_logging_for_logout_request(logout_response, logout_request)
-
-      if errors.size.positive?
+      if saml_response.valid?
+        logout_request  = SingleLogoutRequest.find(logout_response&.in_response_to)
+        if logout_request.present?
+          Rails.logger.info("SLO callback response to '#{logout_response&.in_response_to}' for originating_request_id "\
+            "'#{originating_request_id}'")
+        else
+          Rails.logger.info('SLO callback response could not resolve logout request for originating_request_id '\
+            "'#{originating_request_id}'")
+        end
+      else
+        Rails.logger.info("SLO callback response invalid for originating_request_id '#{originating_request_id}'")
         extra_context = { in_response_to: logout_response&.in_response_to }
         log_message_to_sentry("SAML Logout failed!\n  " + errors.join("\n  "), :error, extra_context)
       end
-    rescue => e
+    rescue StandardError => e
       log_exception_to_sentry(e, {}, {}, :error)
     ensure
-      logout_request&.destroy
```

Comment:
no need to destroy because the ttl is short?

Reply:
no need to destroy because who cares really? it's going to go away and get replaced by SessionActivity

Reply:
added it back.